docs/dev/setup/kubernetes.rst

```diff
@@ -0,0 +1,288 @@
+
+.. figure:: kubernetes/rancher_nav_workloads.png
+   :align: center
+
+
+.. figure:: kubernetes/rancher_empty_workloads.png
+   :align: center
+
+6. Create new namespace in Rancher
+
+Namespaces are virtual clusters backed by the same physical cluster. Namespaces provide a scope for names. Names of resources need to be unique within a namespace, but not across namespaces.
+Usually different namespaces are created to separate environments deployments i.e. development, staging, production.
+
+For our development purposes we will create a namespace called artemis.
+It can be done easily using Rancher.
+
+a. Navigate to Namespaces using the top menu of Rancher
+
+b. Select ``Add Namespace`` to open the form for namespace creation  
+
+   .. figure:: kubernetes/rancher_namespaces.png
+      :align: center
+
+c. Put ``artemis`` as namespace's name and select the ``Create`` button
+
+   .. figure:: kubernetes/rancher_create_namespace.png
+      :align: center
+
+
+
+Create DockerHub repository
+---------------------------
+The Artemis image will be stored and managed in DockerHub. Kubernetes will pull it from there and deploy it afterwards.
+After you log in to your `DockerHub <https://hub.docker.com/>`__ account you can create as many public repositories as you want.
+In order to create a repository you need to select the ``Create repository`` button.
+
+
+**DockerHub:**
+
+.. figure:: kubernetes/dockerhub.png
+   :align: center
+
+Then fill in the repository name with ``artemis``. The use the ``Create`` button and you repository is created.
+
+.. figure:: kubernetes/dockerhub_create_repository.png
+   :align: center
+
+Configure Docker ID (username)
+------------------------------
+The username in DockerHub is called Docker ID. You need to seet your Docker ID in the ``artemis-deployment.yml`` resource so that Kubernetes knows where to pull the image from.
+Open the ``src/main/kubernetes/artemis-k8s/artemis-deployment.yml`` file and edit
+
+   ::
+
+      template:
+         spec:
+         containers:
+            image: <DockerId>/artemis
+
+and replace <DockerId> with your docker ID in DockerHub
+
+i.e. it will look like this:
+
+   ::
+
+      template:
+         spec:
+         containers:
+            image: mmehmed/artemis
+
+
+
+Configure Artemis resources
+---------------------------
+In order to run Artemis, you need to configure the Artemis resources with the configuration you are going to use it with i.e. Jira, Bitbucket, Bamboo or Jenkins, Gitlab.
+Make sure you have configured the ``src/main/resources/config/application-prod.yml`` or ``src/main/resources/config/application-artemis.yml`` file with the proper configuration. 
+
+Since the deployment is done on a Kubernetes cluster, localhost connections to Jira, Bamboo, Bitbucket or Gitlab, Jenkins will not work. 
+For this reason you should set the connection to existing servers or to local Kubernetes deployments.
+
+For more information check out the following documentations: 
```

Comment:
I followed all the previous steps up to this point. No issues at all. I am even running it on a [WSL](https://docs.microsoft.com/en-us/windows/wsl/about) on Windows which usually is trickier. All ran smoothly, and was able to do it fairly quickly. 👍 

However, I stumbled on this particular section, as I am not sure what to do, or how much is enough to do to get through to the next step underneath.

For example:
> Make sure you have configured the ``src/main/resources/config/application-prod.yml`` or ``src/main/resources/config/application-artemis.yml`` file with the proper configuration.

What is the 'proper configuration'. I am a bit scared to move forward as I don't know how much is proper and how much is not. I would expect some more context here. I assume someone very familiar with those configurations wouldn't ask this trivial question but for someone without the context an explanation would be more fair here. 

> Since the deployment is done on a Kubernetes cluster, localhost connections to Jira, Bamboo, Bitbucket or Gitlab, Jenkins will not work. 
> For this reason you should set the connection to existing servers or to local Kubernetes deployments.

I understand the overall idea of this part, but very unsure to move forward. I want to 'set the connection to existing servers or to local Kubernetes deployments' but I am thinking I will most likely not get it right. Would it make sense to add an example here or screenshots, if it makes sense. 


> For more information check out the following documentations: 

This alone made it even more confusing to me. Do I need to just read everything on those links and come back here again, or should I follow every single step in both/either of the links. If I need to do the whole setup for say Jenksins and Gitlab that setup alone is even longer then all the kubernetes setup x2, I would expect a statment to say: follow all the steps on setting up Jenkins and Gitlab/Bitbucket et.al , in order to continue then with the next steps that is 'Build Artemis'

All in all I found this section quite confusing. 
I would expect more something like the following:

> Because of a certain reason X you need to first set up Y. You can set up Y by following the steps as described [here](link-to-somewhere). Then after setting up Y, you need to do Z because of a certain reason M. In order to do Z follow the steps decribed [here](link-to-somewhere)